src/geodesy.jl

```diff
@@ -0,0 +1,447 @@
+                    output[ii] = max(output[curr_idx],mask[ii])
+                    enqueue!(queue, ii)
+                end
+            end
+        end
+    end
+    return output
+end
+
+#specialization for binary case
+function overbuild(marker::AbstractArray{<:Union{Bool,AbstractGray{Bool}},N}, mask::AbstractArray{<:Union{Bool,AbstractGray{Bool}},N}, connectivity::AbstractArray{Bool})  where {N}
+    marker_inv =  map(!, marker)
+    mask_inv = map(!, mask)
+    output_inv= underbuild(marker_inv,mask_inv,connectivity)
+    return map(!, output_inv)
+end
+
+"""
+    regional_maxima(image, connectivity=[]) -> Array
+
+Determines all *regional maxima* of the `image`.
+
+# Details
+The *regional maximum* is defined as the connected set of pixels that have the
+same value, which is greater than the values of all pixels in direct
+neighborhood of the set.
+This implementation is faster than maxtree approach if maxtree not precomputed
+# Arguments
+- `image::AbstractArray{T, N}`: where {T<:ImageCore.NumberLike , N} the ``N``-dimensional input image
+- `connectivity::AbstractArray{Bool}`: the neighborhood connectivity.
+"""
+regional_maxima(image::AbstractArray{T, N}, connectivity::AbstractArray{Bool}) where {T<:ImageCore.NumberLike , N} = extract_regional_extrema(image, connectivity, isgreater, isequal ) 
+
+"""
+    regional_minima(image, connectivity=[]) -> Array
+
+Determines all *regional minima* of the `image`.
+
+# Details
+The *regional minima* is defined as the connected set of pixels that have the
+same value, which is lower than the values of all pixels in direct
+neighborhood of the set.
+
+This implementation is faster than maxtree approach if maxtree not precomputed
+
+# Arguments
+- `image::AbstractArray{T, N}`: where {T<:ImageCore.NumberLike , N} the ``N``-dimensional input image
+- `connectivity::AbstractArray{Bool}`: the neighborhood connectivity.
+"""
+regional_minima(image::AbstractArray{T, N}, connectivity::AbstractArray{Bool}) where {T<:ImageCore.NumberLike , N} = extract_regional_extrema(image, connectivity, isless, isequal ) 
+
+
+function neighborhood(A::AbstractArray{T,N}, connectivity::AbstractArray{Bool}) where {T,N}
+    all(in((1,3)), size(connectivity)) || throw(ArgumentError("connectivity must have size 1 or 3 in each dimension"))
+    for d = 1:ndims(connectivity)
+        size(connectivity, d) == 1 || reverse(connectivity; dims=d) == connectivity || throw(ArgumentError("connectivity must be symmetric"))
+    end
+    center = CartesianIndex(map(axes(connectivity)) do ax
+        (first(ax) + last(ax)) ÷ 2
+    end)
+    offsets = CartesianIndex{N}[]
+    for i in CartesianIndices(connectivity)
+        if i == center && continue # we don't need to check center
+        end
+        if connectivity[i]
+            push!(offsets, i - center)
+        end
+    end
+    return (offsets...,)   # returning as a tuple allows specialization
```

Comment:
This trick might not work here because it's not type-stable: the length of the tuple depends on the runtime information, i.e., the value of `connectivity`. Thus the return type can't be inferred at compile time.

Reply:
Ok my bad i just copy paste over code without thinking much on it